src/core/lib/resource_quota/arena.cc

```diff
@@ -86,11 +91,9 @@ void Arena::DestroyManagedNewObjects() {
   }
 }
 
-void Arena::Destroy() {
-  DestroyManagedNewObjects();
-  memory_allocator_->Release(total_allocated_.load(std::memory_order_relaxed));
+void Arena::Destroy() const {
   this->~Arena();
-  gpr_free_aligned(this);
+  gpr_free_aligned(const_cast<Arena*>(this));
```

Comment:
Any reason we can't move this to the dtor?  If we do that and use the `UnrefCallDtor` unref behavior, then we can remove both the `Destroy()` method and the `arena_detail::UnrefDestroy` struct.

Reply:
Deallocation and destruction are different things, and it will break weirdly if arena ever picks up a parent class.